### ISSUE TITLE
Butterknife bind(object, view) -> return the view?

### ISSUE THREAD

**Issue by mlostekk** [JakeWharton/butterknife#305]
Hello,

wouldn't it be cool to have the 

    public static void bind(Object target, View source)

more like this (return the view given as parameter)

    public static View bind(Object target, View source)

in order to use it in fragments like that 

    return ButterKnife.bind(this, inflater.inflate(R.layout.fragment, container, false));

instead of 

    View view = inflater.inflate(R.layout.fragment, container, false);
    ButterKnife.bind(this, view);
    return view;

What do you think?

**Comment by JakeWharton** [comment]
This would be a binary incompatible change so it would have to wait until v8.0. I'm not opposed to it, but you can write your own static method which does it for now:
```java
public static <V extends View> V bind(Object o, V view) {
  ButterKnife.bind(o, view);
  return view;
}
```

**Comment by ewintory** [comment]
@JakeWharton Why is this a binary incompatible change?

**Comment by JakeWharton** [comment]
Return type is part of the method signature baked into the class file. See: http://docs.oracle.com/javase/specs/jls/se7/html/jls-8.html#jls-8.4.2

**Comment by JakeWharton** [comment]
Whoops. I meant: http://docs.oracle.com/javase/specs/jls/se7/html/jls-13.html#jls-13.4.15

**Comment by ewintory** [comment]
Got it, thank you :thumbsup:

**Comment by mlostekk** [comment]
sure. wrote my own method already. just thought I suggest it for some future versions!

but in general! keep up the awesome work! (should I close this ticket?)

**Comment by svenjacobs** [comment]
Just curious in what situation someone would update a third party library without recompiling the Android project? Isn't that mandatory on Android because of dexing etc?

**Comment by ugommirikwe** [comment]
He possibly extended the `ButterKnife` class and "overrode" the `bind` method. So in his code he now uses his own ButterKnife subclass in place of the one from the lib.

Am I right, @mlostekk ?

**Comment by JakeWharton** [comment]
@svenjacobs if another library you included was compiled with 7.0.1 and we changed this in 7.0.2 or a 7.1.0 then you would get compilation failures.

**Comment by mlostekk** [comment]
Actually I am having exactly the proxy helper function as mentioned above

	public static <V extends View> V bindButterknife(Object object, V view) 
	{
		ButterKnife.bind(object, view);
		return view;
	}
